test/hotspot/jtreg/runtime/os/TestWXHealing.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2025 IBM Corporation. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @requires os.family == "mac"
+ * @requires os.arch == "aarch64"
+ * @summary Run shell with -XX:+StressWXHealing. This tests most of
+ * the triggers for WX mode.
+ * @library /test/lib
+ * @run main/othervm -XX:+UnlockDiagnosticVMOptions -XX:+TraceWXHealing -XX:+StressWXHealing TestWXHealing
+ * @run main/othervm -Xcomp -Xbatch -XX:+UnlockDiagnosticVMOptions -XX:+TraceWXHealing -XX:+StressWXHealing TestWXHealing
+ */
+
+import jdk.jshell.tool.*;
+
+public class TestWXHealing {
+
+    // There's nothing special about jshell here: we just need an
+    // application that does a lot of compilation and class loading.
+    public static void main(String[] args) throws Throwable {
+        JavaShellToolBuilder
+            .builder()
+            .start();
```

Comment:
Should we count "Healing WXMode" output lines to make sure healing is happening?

Reply:
Never mind, I've done it now.